gammapy/datasets/map.py

```diff
@@ -1548,7 +1556,22 @@ def to_region_map_dataset(self, region, name=None):
             )
 
         if self.exposure:
-            kwargs["exposure"] = self.exposure.to_region_nd_map(region, func=np.mean)
```

Comment:
In `to_region_nd_map` we have :
https://github.com/gammapy/gammapy/blob/92a6eb9f440aa92c73269edcffe8bad4d6493662/gammapy/maps/wcs/ndmap.py#L555-L557
these lines should handle correctly the case where the region do not fully overlap with the dataset.

For me the main difference is that here the function is simply `np.mean` while you use `np.sum()` weighted by solid angle. You can probably define  `weights = self.exposure.geom.solid_angle().self.exposure.geom.solid_angle().data.sum(axis=(1,2))`
 and the masking should be handled correctly by `to_region_nd_map`.

Reply:
> You can probably define weights = self.exposure.geom.solid_angle().self.exposure.geom.solid_angle().data.sum(axis=(1,2))
and the masking should be handled correctly by to_region_nd_map.

This is basically what I did right? With a few more lines of code of course, I agree my implementation can be simplified... But the idea should be the same

Reply:
yes,  but if the weight correction is done  after the application of the region mask within `to_region_nd_map` it would probably be simpler.